sds_data_manager/stacks/ialirt_processing_stack.py

```diff
@@ -0,0 +1,63 @@
+"""Processing Stack
+This is the module containing the general stack to be built for
+computation of I-ALiRT algorithms
+"""
+from aws_cdk import Stack
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import aws_ecr as ecr
+from constructs import Construct
+
+from sds_data_manager.constructs.ialirt_compute_resources import (
+    IalirtEC2Resources,
+)
+from sds_data_manager.stacks import ialirt_dynamodb_stack
+
+
+class IalirtProcessing(Stack):
+    """A processing system for ialirt."""
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        vpc: ec2.Vpc,
+        repo: ecr.Repository,
+        **kwargs,
+    ) -> None:
+        """Constructor
+
+        Parameters
+        ----------
+        scope : Construct
+            Parent construct.
+        construct_id : str
+            A unique string identifier for this construct.
+        vpc : ec2.Vpc
+            VPC into which to put the resources that require networking.
+        repo : ecr.Repository
+            ECR repository containing the Docker image.
+        """
+        super().__init__(scope, construct_id, **kwargs)
+
+        # Setup the EC2 resources
+        self.ec2_resources = IalirtEC2Resources(
+            self,
+            "IalirtEC2Environment",
+            vpc=vpc,
+            repo=repo,
+        )
+
+        # I-ALiRT IOIS DynamoDB
```

Comment:
I think what I would prefer to do is make the dynamodb resource into a construct and take the DynamoDB stack out of the IalirtProcessing stack altogether since we will have multiple DynamoDB tables. I also think that keeping the compute resource as a stack improves modularization.

Reply:
> I think what I would prefer to do is make the dynamodb resource into a construct

I'm not sure I'm completely following this. That is what AWS gives us already (an L2 construct). My main concern is we don't need to write a wrapper around AWS's wrapper unless it greatly simplifies things because that will make it harder to follow what is set where rather than going to AWS' documentation. So if you need multiple tables, just call the `dynamo_db.Table()` constructor multiple times in the various stacks where you want them.

> I also think that keeping the compute resource as a stack improves modularization.

Maybe listing out what resources you are planning in each of the stacks would help me follow. But, I-ALiRT is only a few resources total, which is why I wasn't sure we would need modularization (is this going to be re-used anywhere else, is it nice to deploy standalone, etc.) I was thinking this would be nice to just `cdk deploy i-alirt` rather than `cdk deploy i-alirt-compute i-alirt-storage i-alirt-networking ...`

Reply:
The resources expected is something like this (although I'm sure this list will change some). I would rather start with really modular code and move towards one Stack if we think that makes sense than the opposite.
- ECS w/ autoscaling (maybe some type of load balancing...I'm not sure)
- ECR
- s3 bucket for mounting to container
**- Eventbridge for populating DynamoDB for IOIS binary blob 
- lambda for populating IOIS DynamoDB 
- DynamoDB for IOIS **
- lambda for parsing packet data
- DynamoDB for parsed packet data (table for each instrument?)
- Eventbridge for kicking off next lambda once previous lambda is complete
- lambda: check for complete dataset before applying algorithms; apply algorithms (separate lambda for each instrument?)
- DynamoDB for final data product (table for each instrument?)
- s3 bucket for final data product

bolded = maybe not needed if we are archiving CCSDS packets in s3

Reply:
> I would rather start with really modular code and move towards one Stack if we think that makes sense than the opposite.

It would be good to get other eyes/thoughts on this, because I would go the other way. Start out with something contained and then expand later. It is much easier to grow scope than to shrink IMO.

The one I'm really thinking about is the DynamoDB table Stack. That just seems like you should hard-code it right into here. Currently, I have to go over to this project's DynamoDB constructor to figure out what everything means, and that just calls the aws_dynamodb table constructor, so there is an indirection and the options aren't used yet. I'd rather go to AWS' documentation and figure out what the defaults are.

I think that entire Stack class could be put inline here in ~15 lines of code.
```python
ialirt_table = dynamodb.Table(
            self,
            "DynamoDB-ialirt",
            table_name="ialirt-packets",
            partition_key=dynamodb.Attribute(
                name="ingest-ugps", type=dynamodb.AttributeType.STRING
            ),
            sort_key=dynamodb.Attribute(
                name="sct-vtcw", type=dynamodb.AttributeType.STRING
            ),
            # on-demand
            billing_mode=dynamodb.BillingMode.PAY_PER_REQUEST,
            removal_policy=RemovalPolicy.DESTROY,
            point_in_time_recovery=True,
        )
```

Then, if later you want to add in the read/write units thats OK and should be simple to add, but we aren't using those now so it adds confusion IMO.

Reply:
I like this breakdown. It looks cleaner. I wonder if we should do same in `stackbuilder.py` it's getting crowded.
```
class IalirtProcessing(Stack):
    def __init__(...)
        super().__init__(scope, construct_id, **kwargs)
        self.add_compute_resources()
        self.add_dynamodb_table()

    def add_compute_resources(self, ...):
        """Add EC2 components"""

    def add_dynamodb_table(self, ...):
        """Add DynamoDB Table here"""
```